test/utils.js

```diff
@@ -12,177 +12,179 @@
-        expect(memoized(1)).to.eventually.equal(1),
-        expect(memoized(2)).to.eventually.equal(2),
-        expect(memoized(2)).to.eventually.equal(2),
-      ]).then(() =>
-        expect(makePromise.count).to.equal(2),
-      );
-    });
-
-    it('should not cache errors', () => {
-      const makePromise = counted(err => Promise.reject(err));
-      const memoized = promiseMemoize(makePromise);
-      // Make sure to do these in strict sequence, as memoized DOES cache
-      // promises that aren't in a resolved state yet
-
-      return Promise.resolve()
-          .then(() => expect(memoized(1)).to.be.rejectedWith(1))
-          .then(() => expect(memoized(1)).to.be.rejectedWith(1))
-          .then(() => expect(memoized(2)).to.be.rejectedWith(2))
-          .then(() => expect(memoized(2)).to.be.rejectedWith(2))
-          .then(() => expect(makePromise.count).to.equal(4));
-    });
-    // Needs test: it should cache in-progress promises
-  });
-
-  describe('cleanupingPromise', () => {
-    it('should run cleanup functions when resolved', () => {
-      let value = 0;
-
-      return cleanupingPromise((resolve, reject, cleanup) => {
-        value += 1;
-        cleanup(() => value -= 1);
-        expect(value).to.equal(1);
-
-        value += 1;
-        cleanup(() => value -= 1);
-        expect(value).to.equal(2);
-        resolve(null);
-      }).then(() => {
-        expect(value).to.equal(0);
-      });
-    });
-
-    it('should run cleanup functions if an error is thrown', () => {
-      let value = 0;
-
-      return cleanupingPromise((resolve, reject, cleanup) => {
-        cleanup(() => value += 1);
-        cleanup(() => value += 1);
-        throw new Error();
-      }).catch(e =>
-        expect(value).to.equal(2)
-      );
-    });
-
-    it('should not allow asynchronous cleanups', () => {
-      let value = 0;
-      return cleanupingPromise((resolve, reject, cleanup) => {
-        cleanup(() => value += 1);
-        resolve(Promise.resolve().then(() =>
-          expect(() => cleanup(() => value += 1)).to.throw()
-        ));
-        cleanup(() => value += 1);
-        cleanup(() => value += 1);
-      }).then(() =>
-        expect(value).to.equal(3)
-      );
-    });
-
-    it('should propogate cleanup errors', () => {
-      return expect(cleanupingPromise((resolve, reject, cleanup) => {
-        cleanup(() => {});
-        cleanup(() => {
-          throw new Error();
-        });
-        cleanup(() => {});
-        resolve('SUCCESS');
-      })).to.be.rejected;
-    });
-
-    // Needs test:
-    // - it should wait for cleanup promises
-    // - it shoudl execute all cleanups concurrently, if they're promises
-  });
-});
+// TODO - Add these tests back in once we have typescript supported in our
+// testing pipeline.
+
+/* import {expect, use} from 'chai';
+ * import chaiAsPromised from 'chai-as-promised';
+ * import promiseFinally from 'promise.prototype.finally'; */
+
```

Comment:
Perhaps it would be better to remove the unused code and leave a placeholder rather than comment it out.

Reply:
Good catch, will update.